### ISSUE TITLE
many ANR errors

### ISSUE THREAD

**Issue by KORuL** [Instabug/Instabug-Android#217]
### Steps to Reproduce the Problem
Too many ANR errors
### Instabug integration code
*Make sure to remove your app token.*

### SDK Version
    `implementation 'com.instabug.library:instabug:8.6.0'`
### [Optional] Project That Reproduces the Issue
![image](https://user-images.githubusercontent.com/19871487/63297704-a8133180-c2da-11e9-909f-415a2e2b444c.png)

**Comment by mzelzoghbi** [comment]
Hey @KORuL, Thanks for reporting this, can you provide us with the ANR stack traces?

**Comment by KORuL** [comment]
![image](https://user-images.githubusercontent.com/19871487/63298365-f6750000-c2db-11e9-8a58-39628b349057.png)
![image](https://user-images.githubusercontent.com/19871487/63298388-042a8580-c2dc-11e9-9c11-f8fd2d67822f.png)
![image](https://user-images.githubusercontent.com/19871487/63298409-0ee51a80-c2dc-11e9-9540-3e08ba4958dc.png)
![image](https://user-images.githubusercontent.com/19871487/63298429-1c020980-c2dc-11e9-84c7-3df7caa1b155.png)
![image](https://user-images.githubusercontent.com/19871487/63298453-28866200-c2dc-11e9-8365-3365e407977f.png)
![image](https://user-images.githubusercontent.com/19871487/63298476-32a86080-c2dc-11e9-9a8c-8626e6200956.png)

**Comment by mzelzoghbi** [comment]
Hey @KORuL 

We created a snapshot that should fix the issue, Can you please try it and let us know if everything works out?

```
implementation 'com.instabug.library:instabug:8.6.1.1-SNAPSHOT '
```
And don't forget to add this to your app **build.gradle** before your dependencies
```
repositories {
    ....
    maven {
        url "https://oss.sonatype.org/content/repositories/snapshots"
    }
    ...
}
```

**Comment by KORuL** [comment]
Yes, I try it tommorow

**Comment by KORuL** [comment]
too much anr and in the new version. people start to get scared of it. How do you want me to pay more than $ 100 per month - if people leave because of your sdk?
If anr does not disappear, I will have to go to your competitors (although I really like you.

**Comment by KORuL** [comment]
![Screenshot_2019-08-23-11-55-47-847_com google android apps playconsole](https://user-images.githubusercontent.com/19871487/63580447-85fd0600-c59d-11e9-9e0a-481e70c28037.png)
![Screenshot_2019-08-23-11-55-54-020_com google android apps playconsole](https://user-images.githubusercontent.com/19871487/63580448-85fd0600-c59d-11e9-8519-8c512fdb1ea5.png)
![Uploading Screenshot_2019-08-23-11-55-25-292_com.google.android.apps.playconsole.png…]()
![Uploading Screenshot_2019-08-23-11-55-29-818_com.google.android.apps.playconsole.png…]()
![Uploading Screenshot_2019-08-23-11-55-59-876_com.google.android.apps.playconsole.png…]()
![Uploading Screenshot_2019-08-23-11-55-18-031_com.google.android.apps.playconsole.png…]()

**Comment by fchicken** [comment]
Hi @mzelzoghbi,

I have many ANR problems too.
Here are part of the stack traces.

#### executing service live.free.tv_tw/com.instabug.library.network.worker.uploader.InstabugSessionUploaderService
```
"main" tid=1 Blocked 
"main" prio=5 tid=1 Blocked
  | group="main" sCount=1 dsCount=0 obj=0x73ea5ea0 self=0xb72fd238
  | sysTid=11950 nice=-4 cgrp=default sched=0/0 handle=0xb6f27b34
  | state=S schedstat=( 0 0 0 ) utm=118 stm=36 core=1 HZ=100
  | stack=0xbe406000-0xbe408000 stackSize=8MB
  | held mutexes=
  at com.instabug.library.b.b (b.java)
- waiting to lock <0x0144eb52> (a com.instabug.library.b) held by thread 9
  at com.instabug.library.network.b.onReceive (b.java:37)
  at android.app.LoadedApk$ReceiverDispatcher$Args.run (LoadedApk.java:915)
  at android.os.Handler.handleCallback (Handler.java:739)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:168)
  at android.app.ActivityThread.main (ActivityThread.java:5885)
  at java.lang.reflect.Method.invoke! (Native method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:797)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:687)

"AsyncTask #1" tid=9 Native 正在執行網路 I/O
"AsyncTask #1" prio=5 tid=9 Native
  | group="main" sCount=1 dsCount=0 obj=0x12c28340 self=0xb74fc9e0
  | sysTid=11972 nice=10 cgrp=bg_non_interactive sched=0/0 handle=0x9f02c930
  | state=S schedstat=( 0 0 0 ) utm=32 stm=3 core=0 HZ=100
  | stack=0x9ef2a000-0x9ef2c000 stackSize=1038KB
  | held mutexes=
  #00  pc 000000000004359c  /system/lib/libc.so (__ppoll+20)
  #01  pc 000000000001b1cd  /system/lib/libc.so (poll+32)
  #02  pc 0000000000022b39  /system/lib/libjavacore.so (???)
  #03  pc 000000000045236d  /system/framework/arm/boot.oat (Java_libcore_io_Posix_poll___3Landroid_system_StructPollfd_2I+104)
  at libcore.io.Posix.poll (Native method)
  at libcore.io.BlockGuardOs.poll (BlockGuardOs.java:195)
  at libcore.io.IoBridge.isConnected (IoBridge.java:215)
  at libcore.io.IoBridge.connectErrno (IoBridge.java:171)
  at libcore.io.IoBridge.connect (IoBridge.java:122)
  at java.net.PlainSocketImpl.connect (PlainSocketImpl.java:183)
  at java.net.PlainSocketImpl.connect (PlainSocketImpl.java:452)
  at java.net.Socket.connect (Socket.java:884)
- locked <0x04ce02fc> (a java.lang.Object)
  at com.android.okhttp.internal.Platform.connectSocket (Platform.java:117)
  at com.android.okhttp.internal.http.SocketConnector.connectRawSocket (SocketConnector.java:160)
  at com.android.okhttp.internal.http.SocketConnector.connectTls (SocketConnector.java:79)
  at com.android.okhttp.Connection.connect (Connection.java:143)
  at com.android.okhttp.Connection.connectAndSetOwner (Connection.java:185)
  at com.android.okhttp.OkHttpClient$1.connectAndSetOwner (OkHttpClient.java:128)
  at com.android.okhttp.internal.http.HttpEngine.nextConnection (HttpEngine.java:341)
  at com.android.okhttp.internal.http.HttpEngine.connect (HttpEngine.java:330)
  at com.android.okhttp.internal.http.HttpEngine.sendRequest (HttpEngine.java:248)
  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute (HttpURLConnectionImpl.java:457)
  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponse (HttpURLConnectionImpl.java:408)
  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponseCode (HttpURLConnectionImpl.java:540)
  at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getResponseCode (DelegatingHttpsURLConnection.java:105)
  at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getResponseCode (HttpsURLConnectionImpl.java:25)
  at com.instabug.library.network.NetworkManager$1.a (NetworkManager.java:117)
  at io.reactivex.c.e.b.c.b (ObservableCreate.java:40)
  at io.reactivex.h.a (Observable.java:12025)
  at com.instabug.library.network.a.a$1.run (FeaturesService.java:1074)
  at com.instabug.library.b.b (InstabugFeaturesManager.java:2022)
- locked <0x0144eb52> (a com.instabug.library.b)
  at com.instabug.library.network.worker.a.a.onHandleTask (FeaturesFetcherTask.java:1021)
  at com.instabug.library.network.BaseNetworkTask$a.a (BaseNetworkTask.java:49)
  at com.instabug.library.network.BaseNetworkTask$a.doInBackground (BaseNetworkTask.java:39)
  at android.os.AsyncTask$2.call (AsyncTask.java:295)
  at java.util.concurrent.FutureTask.run (FutureTask.java:237)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1113)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at java.lang.Thread.run (Thread.java:818)
```

#### executing service live.free.tv_tw/com.instabug.library.analytics.network.InstabugAnalyticsUploaderService
```
"main" tid=1 Blocked 
"main" prio=5 tid=1 Blocked
  | group="main" sCount=1 dsCount=0 obj=0x75b50698 self=0x7fa9e7ba00
  | sysTid=8413 nice=-4 cgrp=default sched=0/0 handle=0x7fad538fc8
  | state=S schedstat=( 2793132999 632855954 4498 ) utm=197 stm=82 core=4 HZ=100
  | stack=0x7fdaae7000-0x7fdaae9000 stackSize=8MB
  | held mutexes=
  at com.instabug.library.b.b (b.java)
- waiting to lock <0x0e76a471> (a com.instabug.library.b) held by thread 19
  at com.instabug.library.network.b.onReceive (b.java:37)
  at android.app.LoadedApk$ReceiverDispatcher$Args.run (LoadedApk.java:957)
  at android.os.Handler.handleCallback (Handler.java:739)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:179)
  at android.app.ActivityThread.main (ActivityThread.java:5730)
  at java.lang.reflect.Method.invoke! (Native method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:791)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:681)
```
but no tid=19 found in stack traces

#### executing service live.free.tv_tw/com.instabug.crash.network.InstabugCrashesUploaderService
```
"main" prio=5 tid=1 Blocked
  | group="main" sCount=1 dsCount=0 obj=0x7556efb8 self=0x556807fbe0
  | sysTid=14070 nice=-4 cgrp=default sched=0/0 handle=0x7f9792d2c0
  | state=S schedstat=( 0 0 0 ) utm=93 stm=25 core=5 HZ=100
  | stack=0x7fda244000-0x7fda246000 stackSize=8MB
  | held mutexes=
  at com.instabug.library.b.b (b.java)
- waiting to lock <0x0e252c61> (a com.instabug.library.b) held by thread 87
  at com.instabug.library.network.b.onReceive (b.java:37)
```
but no tid=87 found in stack traces

There are still many other issues with similar stack traces.
Please let me know if you need more!

**Comment by mzelzoghbi** [comment]
Hi @KORuL, we are very sorry for this bad experience and we are working hard on getting this fixed, we need some help from you by providing detailed info for the new ANRs in the screenshots.

Thanks

**Comment by mzelzoghbi** [comment]
Hi @fchicken , Thanks for reporting this. Can you please try the snapshot mentioned here https://github.com/Instabug/Instabug-Android/issues/217#issuecomment-522750958 and tell us if it worked 

Thanks

**Comment by mzelzoghbi** [comment]
Hello everyone, We just released **8.6.2** that fixes the ANR issues. Please feel free to reopen this issue if you encounter the same issue again. Thanks
